hudi-client/src/test/java/org/apache/hudi/common/HoodieTestDataGenerator.java

```diff
@@ -80,16 +79,27 @@
       + "{\"name\": \"begin_lat\", \"type\": \"double\"},{\"name\": \"begin_lon\", \"type\": \"double\"},"
       + "{\"name\": \"end_lat\", \"type\": \"double\"},{\"name\": \"end_lon\", \"type\": \"double\"},"
       + "{\"name\":\"fare\",\"type\": \"double\"}]}";
+  public static String GROCERY_PURCHASE_SCHEMA = "{\"type\":\"record\",\"name\":\"purchaserec\",\"fields\":["
```

Comment:
Could be better to re-use the same schema? if you really want two different schema, then its time to modularize this class better?

Reply:
@vinothchandar I wanted to depict a real-life scenario in test-cases. Hence preferred to have different schemas for every topic. By modularizing the class, you mean I should write a class which extends this class?

Reply:
@bvaradar I am still trying to understand why do you want to distinguish the topic from which the record with TestRawTripPayload got ingested? In the current setup also, all the test cases are passing. Could you please make your intention more clear?

Reply:
I think what @bvaradar is trying to get at it is that the current payload is built around the trip schema.. Overall, can we just have multiple topics with the same trips schema? I am not sure what real benefits we get from introducing new schema here.. if we do, I would like to spend more time structuring this class well , since it will make this class even more unclean than what it is today..

Reply:
That last sentence, can happen on a separate JIRA..

Reply:
Ok, got the point. I will get back to you on this @vinothchandar .

Reply:
So as per the design of our updates and inserts generation in HoodieTestDataGenerator class, we are keeping track of number of keys generated corresponding to a particular schema/topic. So the best that I can do is to have multiple topics actually consume the same schema, but the schema variable needs to be different for every topic. This way I will be able to track the number of keys generated or updated for every topic in a better way. You can have a look at HoodieTestDataGenerator class to get a better idea of what I am trying to say. I have defined the following variables now - 

Map<String, Map<Integer, KeyPartition>> existingKeysBySchema;
Map<String, Integer> numKeysBySchema;

 Let me know your thoughts on this @vinothchandar @bvaradar